08-package-metadata.Rmd

```diff
@@ -0,0 +1,362 @@
+
+You can use `usethis::use_*_license()` for configuration.
+
+More in Chapter 9!
+
+**Notes from session**
+
+Jon:
+Be sure to check the license of the source package. Fortunately that's the next chapter! You want to make sure it's a permissive license.
+Still give credit to where you found it in your source code, regardless of how permissive the license might be. For example, I use %||% from {rlang} in a bunch of packages, and, if I'm not importing anything else from {rlang}, I just copy/paste the (very simple) definition and note that I copy/pasted it from {rlang}.
+
+The thing @Collin Berke showed is called re-exporting, and that's different from copy/pasting code. It is completely ok to re-export functions; it still requires the user to have the other package, and it explicitly gives credit to the other package. See usethis::use_pipe() for an example of that. usethis::use_tidy_eval() is another one.
+
+Tan:
+re-exporting also avoids some of the permissive licensing issues!
+
+## Version
+
+- Determines if dependencies are satisfied
+- Two integers and separated by `.` or `-` (recommended to use `.`)
+- Starts at `0.0.0.9000`
+  - Based on Semantic Versioning and by the X.Org versioning schemes
+
+Equivalent examples:
+```
+numeric_version("1.9") == numeric_version("1.9.0")
+```
+
+Can update version number by running `usethis::use_version()` or `usethis::use_dev_version()`.
+
+## Other fields
+
+### `URL` and `BugReports`
+
+- The `URL` field is commonly used to advertise the package’s website and to link to a public source repository
+- `BugReports` is the URL where bug reports should be submitted
+
+`usethis::use_github()` will automatically create these for you; `usethis::use_github_links()` can add them for you.
+
+### `Encoding`
+
+- Required if DESCRIPTION does not consist entirely of ASCII character
+- Default is `Encoding: UTF-8`
+
+### `Collate`
+
+- Controls the order in which files are sourced
+  - Most common if you're using S4 (a system for object oriented programing)
+
+### `LazyData`
+
+- If true, data is lazy loaded so users don't have to run `data()`
+- `usethis::use_data()` does this for you
+
+### `Roxygen*`
+
+`create_package()` creates fields related to the roxygen package
+
+## Final notes
+
+Full list of what can be in the `DESCRIPTION` file is in the [R extensions manual](https://cran.r-project.org/doc/manuals/R-exts.html#The-DESCRIPTION-file).
+
+You can also created your own but CRAN does have constraints
+
+- Only English words
+- Use `Config/` prefix or `Note` suffix
+
+## Meeting Videos
+
+### Cohort 1
+
+`r knitr::include_url("https://www.youtube.com/embed/URL")`
+
+<details>
+<summary> Meeting chat log </summary>
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/vQSYeGwX9yQ" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

Comment:
It looks (from this plus from some of the code blocks) like you copied a rendered md into this Rmd. Or maybe you were trying to sort out how the video gets embedded? You need to put the `vQSYeGwX9yQ` part into `URL` in the `knitr::include_url` piece above.

Everything looks fine (as long as it renders) other than this bit, but it doesn't look like standard Rmarkdown code chunks.

Reply:
Whoops, apologies! I think I just have Quarto on the brain.